kb/graql/executor/ExecutorFactory.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * GRAKN.AI - THE KNOWLEDGE GRAPH
+ * Copyright (C) 2019 Grakn Labs Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package grakn.core.kb.graql.executor;
+
+public interface ExecutorFactory {
+    ComputeExecutor compute();
+    QueryExecutor transactional(boolean infer);
```

Comment:
maybe in the future better naming? why not just call a method that returns a new queryExecutor `buildQueryExecutor()` or just `queryExecutor()`? maybe I dont have enough context, just looks strange

Reply:
same for `createComputeExecutor()`/`buildComputeExecutor()`/`computeExecutor()`